packages/components/src/test/table/UsePresentationTable.test.tsx

```diff
@@ -171,4 +171,101 @@ describe("usePresentationTableWithUnifiedSelection", () => {
+      .returns(async () => ({ content: new Content(descriptor, [item]), size: 1 }));
+
+    const { result } = renderHook(() => usePresentationTableWithUnifiedSelection(initialProps), { wrapper: Wrapper });
+
+    const replaceSpy = sinon.stub(Presentation.selection, "replaceSelection");
+    await waitFor(() => expect(result.current.isLoading).to.be.false);
+
+    const onSelectCallback = result.current.onSelect;
+    onSelectCallback(stringifiedKeys);
+    expect(replaceSpy).to.be.calledOnceWith("UnifiedSelectionContext", {}, validKeys, 1);
+  });
+
+  it("Gets invalid keys and does not pass any to the selection with onSelect", async () => {
+    const keys = ["this is not a valid key", ""];
+    const { result } = renderHook(() => usePresentationTableWithUnifiedSelection(initialProps), { wrapper: Wrapper });
+    const onSelectCallback = result.current.onSelect;
+
+    const replaceSpy = sinon.stub(Presentation.selection, "replaceSelection");
+    onSelectCallback(keys);
+
+    expect(replaceSpy).to.have.been.calledOnceWithExactly("UnifiedSelectionContext", {}, [], 1);
+  });
+
+  it("Gets valid keys for rows that are not loaded and does not pass any to the selection with onSelect", async () => {
+    const stringifiedKeys = [createTestECInstanceKey()].map((key) => JSON.stringify(key));
+
+    const { result } = renderHook(() => usePresentationTableWithUnifiedSelection(initialProps), { wrapper: Wrapper });
+    const onSelectCallback = result.current.onSelect;
+
+    const replaceSpy = sinon.stub(Presentation.selection, "replaceSelection");
+    onSelectCallback(stringifiedKeys);
+
+    expect(replaceSpy).to.have.been.calledOnceWithExactly("UnifiedSelectionContext", {}, [], 1);
+  });
+
+  it("returns an array of selectedRows when keys are passed correctly on selectionChange event", async () => {
+    const keys = new KeySet([createTestECInstanceKey()]);
+    sinon.stub(Presentation.selection, "getSelection").returns(new KeySet([createTestECInstanceKey()]));
+
+    const { descriptor, item } = getMockData();
+    presentationManagerMock.setup(async (x) => x.getContentDescriptor(moq.It.is((options) => options.keys.size === keys.size))).returns(async () => descriptor);
+    presentationManagerMock
+      .setup(async (x) => x.getContentAndSize(moq.It.is((options) => options.keys.size === keys.size)))
+      .returns(async () => ({ content: new Content(descriptor, [item]), size: 1 }));
+
+    const { result } = renderHook(() => usePresentationTableWithUnifiedSelection(initialProps), { wrapper: Wrapper });
+
+    await waitFor(() => expect(result.current.isLoading).to.be.false);
+    Presentation.selection.addToSelection("UnifiedSelectionContext", initialProps.imodel, keys, 1);
+
+    await waitFor(() => expect(result.current.isLoading).to.be.false);
+    const resultAfterAdding = result.current.selectedRows;
+    expect(resultAfterAdding.length).to.be.equal(1);
+  });
```

Comment:
I believe we need multiple variations of this test:
1. Ensure `selectedRows` is correct if unified selection already contains a key of one of the rows when the table is initially loaded.
2. Ensure `selectedRows` is correct when initially unified selection changes after the initial table load. For that, we need to rows in the table. And we have 4 cases:
   - 1 row is selected on initial load, 1 row is added to unified selection using `addToSelection` and we ensure both rows get selected in the table.
   - rows are selected on initial load, 1 row is removed from unified selection using `removeFromSelection` and we ensure only one is selected in the table.
   - row is selected on initial load, we change the selection to the other one using `replaceSelection` and ensure table selection is valid.
   - rows are selected, we clear the selection using `clearSelection` and ensure no rows are selected in the table.